### PR TITLE
ACE3 Implementation

### PR DESCRIPTION
**Pull request by NickvdLee**
- Removes AGM support from F3
- Adds ACE3 Support to F3
  - Supports forcing ACE3 standard or advanced medical system
  - Supports adding unit type appropriate ACE3 items and medical items, depending on medical system setting
  - Supports converting vanilla cargo into cargo from the appropiate medical system
- Fixes missing semicolon in description.ext

### CODE REVIEW

**Comment by cptnnick** [comment]
#615 #616

File: description.ext [hunk 1]
```diff
@@ -173,14 +173,28 @@ class Params
     class f_param_medical
     {
             title = "Medical System";
-            values[] = {0,1,2};
-            texts[] = {"Arma 3 Default","F3 Simple Wounding System","Authentic Gameplay Mod"};
+            values[] = {0,1,2,3};
+            texts[] = {"Arma 3 Default (Autodetect)","F3 Simple Wounding System","Force ACE3 Simple","Force ACE3 Advanced"};
```
Comment (cptnnick) [inline description.ext]: @Sniperhid Pointed out the wording of this might be improved.

File: f/medical/fn_medicalPreInit.sqf [hunk 2]
```diff
@@ -0,0 +1,60 @@
+// F3 - Medical Systems Pre Initialisation
+// Credits: Please see the F3 online manual (http://www.ferstaberinde.com/f3/en/)
+// ====================================================================================
+
+// Wait until medical param is initialized by the pre init param processor
+waitUntil
+{
+	!(isNil {missionNamespace getVariable "f_var_medical"})
+};
```
Comment (cptnnick) [inline f/medical/fn_medicalPreInit.sqf]: Pre-Init is an unscheduled environment, this might need integration with @Sniperhid`s new param preprocessor.

**Comment by cptnnick** [comment]
TODO: 
* Playtest to find the right balance for handing out medical gear
* Documentation

File: f/medical/fn_medicalPreInit.sqf [hunk 3]
```diff
@@ -0,0 +1,61 @@
+// F3 - Medical Systems Pre Initialisation
+// Credits: Please see the F3 online manual (http://www.ferstaberinde.com/f3/en/)
+// ====================================================================================
+
+// Wait until medical param is initialized by the pre init param processor
+// Pre Init is unscheduled space...
+//waitUntil
+//{
+//	!(isNil {missionNamespace getVariable "f_var_medical"})
+//};
```
Comment (cptnnick) [inline f/medical/fn_medicalPreInit.sqf]: @Sniperhid, with your new preprocessor code we can be relatively sure the params will be initialized, right? Or will I need to preprocess the f_var_medical myself here?

Comment (Sniperhid) [inline f/medical/fn_medicalPreInit.sqf]: It will require some testing but I believe there is a good chance we make this happen!

**Comment by whiztler** [comment]
The medic loadout for advanced is insufficient imo. Suggest following loadout:

'''
if (_typeOfUnit == "m") then
{
	// BACKPACK: LIGHT
	if (f_param_backpacks <= 1) then {
		(unitBackpack player) addItemCargoGlobal ["ACE_fieldDressing",  8];
		(unitBackpack player) addItemCargoGlobal ["ACE_elasticBandage",  10];
		(unitBackpack player) addItemCargoGlobal ["ACE_quikclot",  10];
		(unitBackPack player) addItemCargoGlobal ["ACE_packingBandage", 6];
		
		(unitBackpack player) addItemCargoGlobal ["ACE_atropine",  8];
		(unitBackpack player) addItemCargoGlobal ["ACE_morphine", 6];
		(unitBackpack player) addItemCargoGlobal ["ACE_epinephrine",   6];

		(unitBackpack player) addItemCargoGlobal ["ACE_salineIV_250", 4];
		(unitBackpack player) addItemCargoGlobal ["ACE_bloodIV", 2];
		(unitBackpack player) addItemCargoGlobal ["ACE_plasmaIV", 2];
		
		(unitBackpack player) addItemCargoGlobal ["ACE_tourniquet", 3];
		(unitBackpack player) addItemCargoGlobal ["ACE_personalAidKit", 1];
	};
	// BACKPACK: HEAVY
	if (f_param_backpacks == 2) then {
		(unitBackpack player) addItemCargoGlobal ["ACE_fieldDressing",  10];
		(unitBackpack player) addItemCargoGlobal ["ACE_elasticBandage",  12];
		(unitBackpack player) addItemCargoGlobal ["ACE_quikclot",  12];
		(unitBackPack player) addItemCargoGlobal ["ACE_packingBandage", 8];
		
		(unitBackpack player) addItemCargoGlobal ["ACE_atropine",  10];
		(unitBackpack player) addItemCargoGlobal ["ACE_morphine", 8];
		(unitBackpack player) addItemCargoGlobal ["ACE_epinephrine",   8];

		(unitBackpack player) addItemCargoGlobal ["ACE_salineIV_250", 5];
		(unitBackpack player) addItemCargoGlobal ["ACE_bloodIV", 2];
		(unitBackpack player) addItemCargoGlobal ["ACE_plasmaIV", 2];
		
		(unitBackpack player) addItemCargoGlobal ["ACE_tourniquet", 5];
		(unitBackpack player) addItemCargoGlobal ["ACE_personalAidKit", 2];
	};
}; 
'''

**Comment by cptnnick** [comment]
Thank you for your feedback. I don't have a lot of playing experience with
the advanced system.

That's 9 blood/saline packs per medic. How many need to be used per
casualty typically?

**Comment by whiztler** [comment]
Depending on the injury, but medics go thru blood/saline very fast with advanced medical. 

See here for ACE Advanced Medical loadout suggestions:

http://ace3mod.com/wiki/feature/medical-system.html

**Comment by Pasrules** [comment]
Looking at the wiki for ACE would it be worth adding 2 new infantry classes to the assignGear? Combat First Responder and Paramedic would both be classes that i would use when making a mission. I don't mind writing them up if people think this would be worth it.

**Comment by Sniperhid** [comment]
According to @Pasrules this isn't working on dedicated servers at present. Nick we should explore this!

**Comment by cptnnick** [comment]
@Snippers it is working fine for at least one other community (PA). Maybe @Pasrules can share his mission file and I can take a look? @Poulern can you confirm it works?

**Comment by Pasrules** [comment]
Ran a vanilla cptnnick:dev-ace3 on our server with medical parameter set to force ace advanced and the ace medical system does not function as advanced and the advanced medical gear does not spawn.

**Comment by Poulern** [comment]
We have postinit =1 in order for it to work.

**Comment by Pasrules** [comment]
pstInit = 1 fixed the gear loading problem, thanks @Poulern and I emptyed the fn_medicalPreInit.sqf to a blank file. It works, I'll let you guys know if a full missions runs without any problems.

**Comment by Muppet1856** [comment]
Wiki Page for this is missing the enumeration for "Force ACE3 Advanced" and the updated enumeration for "Force ACE3 Simple".  Also, the enumeration only shows 0-2 on the wiki [Medical Systems Support](http://ferstaberinde.com/f3/en//index.php?title=Medical_Systems_Support#Description)

**Pull request closed without merging**